packages/test-devtools-evm-foundry/contracts/TestHelperOz5.sol

```diff
@@ -310,30 +317,97 @@ contract TestHelperOz5 is Test, OptionsHelper {
+     * @param _packetAmount Amount of packets to process.
+     * @param _composer The lzCompose composer address.
+     * @param _expectedReceiveRevertData Expected revert data for lzReceive.
+     * @param _expectedComposeRevertData Expected revert data for lzCompose.
      */
-    function verifyPackets(uint32 _dstEid, address _dstAddress) public {
-        verifyPackets(_dstEid, bytes32(uint256(uint160(_dstAddress))), 0, address(0x0));
+    function verifyAndExecutePackets(
+        uint32 _dstEid,
+        bytes32 _dstAddress,
+        uint256 _packetAmount,
+        address _composer,
+        bytes memory _expectedReceiveRevertData,
+        bytes memory _expectedComposeRevertData
+    ) public {
+        verifyAndExecutePackets(_dstEid, _dstAddress, _packetAmount, _composer, _expectedReceiveRevertData, _expectedComposeRevertData, bytes4(0), bytes4(0));
+    }
+
+     /**
+     * @dev verify and execute packets to destination chain's OApp address.
+     * @param _dstEid The destination endpoint ID.
+     * @param _dstAddress The destination address.
+     * @param _packetAmount Amount of packets to process.
+     * @param _composer The lzCompose composer address.
+     * @param _expectedReceiveRevertData Expected revert data for lzReceive.
+     * @param _expectedComposeRevertData Expected revert data for lzCompose.
+     */
+    function verifyAndExecutePackets(
+        uint32 _dstEid,
+        bytes32 _dstAddress,
+        uint256 _packetAmount,
+        address _composer,
+        bytes4 _expectedReceiveRevertData,
+        bytes4 _expectedComposeRevertData
+    ) public {
+        verifyAndExecutePackets(_dstEid, _dstAddress, _packetAmount, _composer, "", "", _expectedReceiveRevertData, _expectedComposeRevertData);
     }
 
     /**
-     * @dev dst UA receive/execute packets
-     * @dev will NOT work calling this directly with composer IF the composed payload is different from the lzReceive msg payload
+     * @dev verify and execute packets to destination chain's OApp address.
+     * @param _dstEid The destination endpoint ID.
+     * @param _dstAddress The destination address.
+     * @param _packetAmount Amount of packets to process.
+     * @param _composer The lzCompose composer address.
+     * @param _expectedReceiveRevertData Expected revert data for lzReceive in bytes.
+     * @param _expectedComposeRevertData Expected revert data for lzCompose in bytes.
+     * @param _expectedReceiveRevertData4 Expected revert data for lzReceive in bytes4.
+     * @param _expectedComposeRevertData4 Expected revert data for lzCompose in bytes4.
      */
-    function verifyPackets(uint32 _dstEid, bytes32 _dstAddress, uint256 _packetAmount, address _composer) public {
+    function verifyAndExecutePackets(
+        uint32 _dstEid,
+        bytes32 _dstAddress,
+        uint256 _packetAmount,
+        address _composer,
+        bytes memory _expectedReceiveRevertData,
+        bytes memory _expectedComposeRevertData,
```

Comment:
I dont know if I really agree on imbedding the expected revert data into the function like this, is there a functional reason why we dont just rely on the bubbled up revert and catch it with the vm.expectRevert inside of the test files implementation of this contract?

Otherwise it's unconventional compared to how you traditionally would read a testfile using vm.

Reply:
We could probably do that if we expose more granular functions from the TestHelper, eg. split verification and execution of packets. Right now vm.expectRevert only allows to catch revert in next function call or something and it doesnt catch reverts properly

Reply:
Agreed with @DanL0 , you can't just expectRevert() to be a catch-all for `verifyPackets()`